src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.ByteArray.cs

```diff
@@ -25,9 +26,8 @@ public static partial class JsonSerializer
             TValue value,
             JsonSerializerOptions? options = null)
         {
-            Type runtimeType = GetRuntimeType(value);
-            JsonTypeInfo jsonTypeInfo = GetTypeInfo(options, runtimeType);
-            return WriteBytesUsingSerializer(value, jsonTypeInfo);
+            JsonTypeInfo<TValue> jsonTypeInfo = GetTypeInfo<TValue>(options);
```

Comment:
I assume this is the actual change for being more consistent with polymorphism?

Reply:
Correct, we no longer consult the runtime type here, it becomes the responsibility of the converter.